capstone-frontend/src/App.js

```diff
@@ -1,25 +1,41 @@
 import React from 'react';
-import logo from './logo.svg';
+import {
+  Route, 
+  BrowserRouter as Router
+} from 'react-router-dom';
+
 import './App.css';
 
+import Home from './components/Home';
+import Browse from './components/Browse';
+import MyBooks from './components/MyBooks';
+import MyLists from './components/MyLists';
+import MyClubs from './components/MyClubs';
+import BookPage from './components/BookPage';
+import ListPage from './components/ListPage';
+
+import Navbar from './components/Navbar';
+import LeftSideBar from './components/LeftSideBar';
+import RightSideBar from './components/RightSideBar';
+
 function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <Navbar />
+      <div className="row">
+        <LeftSideBar />
+        <div className="col-12 col-md-8" id="body-row">
+          <Route exact path='/' component={Home} />
+          <Route path='/browse' component={Browse} />
+          <Route path='/mybooks' component={MyBooks} />
+          <Route path='/mylists' component={MyLists} />
+          <Route path='/myclubs' component={MyClubs} />
+          <Route path='/bookdetails' render={(props) => <BookPage {...props} />} />
+          <Route path='/listdetails' render={(props) => <ListPage {...props} />} />
```

Comment:
Why `render` instead of `component` here? I think this should be equivalent to `component={BookPage}` or `component={ListPage}` but let me know if that's resulting in different behavior.

Reply:
I believe that this syntax lets us access props passed by the Link that routes a user to a BookPage. For example, a SearchBookItem (boxes rendered by search) would have a Link of the form: 
```
<Link to={{
                    pathname: "/bookdetails", 
                    state: {
                        title: this.props.title,
                        author: this.props.authorName,
                        imgLink: this.props.imgLink,
                        gbook_id: this.props.gbook_id
                    }
                }}>
```
Then, the BookPage can access these props using this.props.location.state.*